### ISSUE TITLE
'app.error.408 Authentication timeout' and redis connection refused

### ISSUE THREAD

**Issue by dev4223** [bigbluebutton/bigbluebutton#13178]
**Describe the bug**
After a fresh install of BBB 2.3.13 with Greenlight 2.9.2 after entering a room I get the white dots and after a time the message "app.error.408 Authentication timeout"

```
BigBlueButton Server 2.3.13 (2543)
                    Kernel version: 4.15.0-156-generic
                      Distribution: Ubuntu 18.04.5 LTS (64-bit)
                            Memory: 65960 MB
                         CPU cores: 16

/etc/bigbluebutton/bbb-web.properties (override for bbb-web)
/usr/share/bbb-web/WEB-INF/classes/bigbluebutton.properties (bbb-web)
       bigbluebutton.web.serverURL: https://bbb.domain.de
                defaultGuestPolicy: ALWAYS_ACCEPT
                 svgImagesRequired: true

/etc/nginx/sites-available/bigbluebutton (nginx)
                       server_name: bbb.domain.de
                              port: 80, [::]:80
                              port: 443 ssl

/opt/freeswitch/etc/freeswitch/vars.xml (FreeSWITCH)
                       local_ip_v4: 188.40.132.242
                   external_rtp_ip: 168.119.176.53
                   external_sip_ip: 188.40.132.242

/opt/freeswitch/etc/freeswitch/sip_profiles/external.xml (FreeSWITCH)
                        ext-rtp-ip: $${local_ip_v4}
                        ext-sip-ip: $${local_ip_v4}
                        ws-binding: 188.40.132.242:5066
                       wss-binding: 188.40.132.242:7443

/usr/local/bigbluebutton/core/scripts/bigbluebutton.yml (record and playback)
                     playback_host: bbb.domain.de
                 playback_protocol: https
                            ffmpeg: 4.2.4-1ubuntu0.1bbb2~18.04

/etc/bigbluebutton/nginx/sip.nginx (sip.nginx)
                        proxy_pass: 188.40.132.242
                          protocol: http

/usr/local/bigbluebutton/bbb-webrtc-sfu/config/default.yml (Kurento SFU)
                        kurento.ip: 188.40.132.242
                       kurento.url: ws://127.0.0.1:8888/kurento
                    kurento.sip_ip: 188.40.132.242
                    localIpAddress: 188.40.132.242
               recordScreenSharing: true
                     recordWebcams: true
                  codec_video_main: VP8
               codec_video_content: VP8

/usr/share/meteor/bundle/programs/server/assets/app/config/settings.yml (HTML5 client)
                             build: 1842
                        kurentoUrl: wss://bbb.domain.de/bbb-webrtc-sfu
                  enableListenOnly: true
                    sipjsHackViaWs: true

/usr/share/bbb-web/WEB-INF/classes/spring/turn-stun-servers.xml (STUN Server)
                              stun: turn.domain.de


# Potential problems described below
```

```
nginx —————————————————► [✔ - active]
freeswitch ————————————► [✔ - active]
redis-server ——————————► [✔ - active]
bbb-apps-akka —————————► [✘ - activating]
bbb-fsesl-akka ————————► [✘ - activating]
mongod ————————————————► [✔ - active]
bbb-html5 —————————————► [✔ - active]
bbb-webrtc-sfu ————————► [✔ - active]
kurento-media-server ——► [✔ - active]
kurento-media-server-8888  [✔ - active]
kurento-media-server-8889  [✔ - active]
kurento-media-server-8890  [✔ - active]
bbb-html5-backend@1 ———► [✔ - active]
bbb-html5-backend@2 ———► [✔ - active]
bbb-html5-frontend@1 ——► [✔ - active]
bbb-html5-frontend@2 ——► [✔ - active]
etherpad ——————————————► [✔ - active]
bbb-web ———————————————► [✔ - active]
```

Following is from `bbb-conf --debug`

```
Sep  8 02:09:56 BG-ubuntu-bbb sh[4692]: Original Exception (Redis::CannotConnectError): Error connecting to Redis on 127.0.0.1:6379 (Errno::ECONNREFUSED)
Sep  8 02:09:56 BG-ubuntu-bbb sh[4692]: /usr/lib/ruby/vendor_ruby/rake/application.rb:186:in `standard_exception_handling'
Sep  8 02:09:56 BG-ubuntu-bbb run-prod.sh[4982]: 2021-09-08T02:09:56.108+02:00 WARN  i.l.c.protocol.ConnectionWatchdog - Cannot reconnect: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:6379
Sep  8 02:09:56 BG-ubuntu-bbb run-prod.sh[4982]: message repeated 2 times: [ 2021-09-08T02:09:56.108+02:00 WARN  i.l.c.protocol.ConnectionWatchdog - Cannot reconnect: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:6379]
Sep  8 02:09:56 BG-ubuntu-bbb run-prod.sh[4982]: 2021-09-08T02:09:56.110+02:00 WARN  i.l.c.protocol.ConnectionWatchdog - Cannot reconnect: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:6379
Sep  8 02:09:56 BG-ubuntu-bbb run-prod.sh[4982]: 2021-09-08T02:09:56.110+02:00 WARN  i.l.c.protocol.ConnectionWatchdog - Cannot reconnect: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:6379
Sep  8 02:09:56 BG-ubuntu-bbb systemd_start_frontend.sh[5124]: #033[31merror#033[39m: uncaughtException: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379
Sep  8 02:09:56 BG-ubuntu-bbb systemd_start_frontend.sh[5124]:     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1137:16) {"error":{"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":6379},"stack":"Error: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1137:16)","exception":true,"date":"Wed Sep 08 2021 02:09:56 GMT+0200 (Central European Summer Time)","process":{"pid":5523,"uid":995,"gid":995,"cwd":"/usr/share/meteor/bundle/programs/server","execPath":"/usr/share/node-v12.16.1-linux-x64/bin/node","version":"v12.16.1","argv":["/usr/share/node-v12.16.1-linux-x64/bin/node","/usr/share/meteor/bundle/main.js","program.json"],"memoryUsage":{"rss":110374912,"heapTotal":60407808,"heapUsed":58362200,"external":24057929}},"os":{"loadavg":[0,0.0146484375,0.048828125],"uptime":1951},"trace":[{"column":16,"file":"net.js","function":"TCPConnectWrap.afterConnect [as oncomplete]","line":1137,"method":"afterConnect [as oncomplete]","native":false}]}
Sep  8 02:09:56 BG-ubuntu-bbb systemd_start_frontend.sh[5124]: #033[31merror#033[39m: uncaughtException: Error: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379
Sep  8 02:09:56 BG-ubuntu-bbb systemd_start.sh[5066]: #033[31merror#033[39m: uncaughtException: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379
Sep  8 02:09:56 BG-ubuntu-bbb systemd_start.sh[5066]:     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1137:16) {"error":{"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":6379},"stack":"Error: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1137:16)","exception":true,"date":"Wed Sep 08 2021 02:09:56 GMT+0200 (Central European Summer Time)","process":{"pid":5536,"uid":995,"gid":995,"cwd":"/usr/share/meteor/bundle/programs/server","execPath":"/usr/share/node-v12.16.1-linux-x64/bin/node","version":"v12.16.1","argv":["/usr/share/node-v12.16.1-linux-x64/bin/node","/usr/share/meteor/bundle/main.js","program.json","NODEJS_BACKEND_INSTANCE_ID=2"],"memoryUsage":{"rss":109273088,"heapTotal":58572800,"heapUsed":55715256,"external":23778278}},"os":{"loadavg":[0,0.0146484375,0.048828125],"uptime":1951},"trace":[{"column":16,"file":"net.js","function":"TCPConnectWrap.afterConnect [as oncomplete]","line":1137,"method":"afterConnect [as oncomplete]","native":false}]}
Sep  8 02:09:56 BG-ubuntu-bbb systemd_start.sh[5066]: #033[31merror#033[39m: uncaughtException: Error: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379
Sep  8 02:09:56 BG-ubuntu-bbb systemd_start_frontend.sh[5100]: #033[31merror#033[39m: uncaughtException: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379
Sep  8 02:09:56 BG-ubuntu-bbb systemd_start_frontend.sh[5100]:     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1137:16) {"error":{"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":6379},"stack":"Error: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1137:16)","exception":true,"date":"Wed Sep 08 2021 02:09:56 GMT+0200 (Central European Summer Time)","process":{"pid":5553,"uid":995,"gid":995,"cwd":"/usr/share/meteor/bundle/programs/server","execPath":"/usr/share/node-v12.16.1-linux-x64/bin/node","version":"v12.16.1","argv":["/usr/share/node-v12.16.1-linux-x64/bin/node","/usr/share/meteor/bundle/main.js","program.json"],"memoryUsage":{"rss":110682112,"heapTotal":60407808,"heapUsed":58204728,"external":24094738}},"os":{"loadavg":[0,0.0146484375,0.048828125],"uptime":1951},"trace":[{"column":16,"file":"net.js","function":"TCPConnectWrap.afterConnect [as oncomplete]","line":1137,"method":"afterConnect [as oncomplete]","native":false}]}
Sep  8 02:09:56 BG-ubuntu-bbb systemd_start_frontend.sh[5100]: #033[31merror#033[39m: uncaughtException: Error: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379
Sep  8 02:09:56 BG-ubuntu-bbb systemd_start.sh[5019]: #033[31merror#033[39m: uncaughtException: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379
Sep  8 02:09:56 BG-ubuntu-bbb systemd_start.sh[5019]:     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1137:16) {"error":{"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":6379},"stack":"Error: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1137:16)","exception":true,"date":"Wed Sep 08 2021 02:09:56 GMT+0200 (Central European Summer Time)","process":{"pid":5537,"uid":995,"gid":995,"cwd":"/usr/share/meteor/bundle/programs/server","execPath":"/usr/share/node-v12.16.1-linux-x64/bin/node","version":"v12.16.1","argv":["/usr/share/node-v12.16.1-linux-x64/bin/node","/usr/share/meteor/bundle/main.js","program.json","NODEJS_BACKEND_INSTANCE_ID=1"],"memoryUsage":{"rss":107900928,"heapTotal":58572800,"heapUsed":56103952,"external":23790605}},"os":{"loadavg":[0,0.0146484375,0.048828125],"uptime":1951},"trace":[{"column":16,"file":"net.js","function":"TCPConnectWrap.afterConnect [as oncomplete]","line":1137,"method":"afterConnect [as oncomplete]","native":false}]}
Sep  8 02:09:56 BG-ubuntu-bbb systemd_start.sh[5019]: #033[31merror#033[39m: uncaughtException: Error: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379
Sep  8 02:10:04 BG-ubuntu-bbb bbb-apps-akka[14674]: Exception in thread "main" java.lang.ExceptionInInitializerError
Sep  8 02:10:04 BG-ubuntu-bbb bbb-apps-akka[14674]: Caused by: com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'http'
Sep  8 02:10:09 BG-ubuntu-bbb bbb-apps-akka[14783]: Exception in thread "main" java.lang.ExceptionInInitializerError
Sep  8 02:10:09 BG-ubuntu-bbb bbb-apps-akka[14783]: Caused by: com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'http'
Sep  8 02:10:45 BG-ubuntu-bbb bbb-fsesl-akka[15064]: Exception in thread "main" java.lang.ExceptionInInitializerError
Sep  8 02:10:45 BG-ubuntu-bbb bbb-fsesl-akka[15064]: Caused by: com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'http'
...
   -- Exceptions found in /var/log/bigbluebutton -- 
/var/log/bigbluebutton/bbb-web.log:2021-09-08T02:09:56.108+02:00 WARN  i.l.c.protocol.ConnectionWatchdog - Cannot reconnect: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:6379
/var/log/bigbluebutton/bbb-web.log:2021-09-08T02:09:56.108+02:00 WARN  i.l.c.protocol.ConnectionWatchdog - Cannot reconnect: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:6379
/var/log/bigbluebutton/bbb-web.log:2021-09-08T02:09:56.108+02:00 WARN  i.l.c.protocol.ConnectionWatchdog - Cannot reconnect: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:6379
/var/log/bigbluebutton/bbb-web.log:2021-09-08T02:09:56.110+02:00 WARN  i.l.c.protocol.ConnectionWatchdog - Cannot reconnect: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:6379
/var/log/bigbluebutton/bbb-web.log:2021-09-08T02:09:56.110+02:00 WARN  i.l.c.protocol.ConnectionWatchdog - Cannot reconnect: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:6379
/var/log/bigbluebutton/bbb-web.log:2021-09-08T02:25:30.204+02:00 WARN  i.l.c.protocol.ConnectionWatchdog - Cannot reconnect: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:6379
/var/log/bigbluebutton/bbb-web.log:2021-09-08T02:25:30.205+02:00 WARN  i.l.c.protocol.ConnectionWatchdog - Cannot reconnect: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:6379
/var/log/bigbluebutton/bbb-web.log:2021-09-08T02:25:30.206+02:00 WARN  i.l.c.protocol.ConnectionWatchdog - Cannot reconnect: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:6379
/var/log/bigbluebutton/bbb-web.log:2021-09-08T02:25:30.207+02:00 WARN  i.l.c.protocol.ConnectionWatchdog - Cannot reconnect: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:6379
/var/log/bigbluebutton/bbb-web.log:2021-09-08T02:25:30.207+02:00 WARN  i.l.c.protocol.ConnectionWatchdog - Cannot reconnect: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:6379
```

I've read the other issues regarding redis server is starting to late, but this is already set up through bbb-install.sh.

**Comment by dev4223** [comment]
Solved it. There where some config files for nginx missing.

**Comment by ffdixon** [comment]
Glad you got it solved -- what files were missing?  Was it an issue with `bbb-install.sh`?

**Comment by dev4223** [comment]
Mh, I really don't know. It was a bit strange. Try to explain:
I came from BBB 2.2. So I had to upgrade Ubuntu and then BBB. I don't wanted to erase the whole system, so maybe this was not so good. But I did it this way: First I backuped all the BBB (config) files. The ```apt remove --purge``` all the BBB-, kms- and kurento-packages. Then I forgot to erase all teh leftover files from BBB. Maybe this was an failure, too.
The ```apt dist-upgrade```and ```do-release-upgrade``` without problems.
Then I ran ```bbb-install.sh``` and there the hazzle begins. It throws errors, that bbb-config could not end it's setup process. Then I realized that I forget to erase the leftover files.
So from the package install process in  ```bbb-install.sh``` I had some new config files e.g. in ```/etc/bigbluebutton/``` with .dpkg-new besides the old ones. 
Ok, backup the newly installed config files and then removed/purged all already installed BBB packages and erased all these leftover files and diretories this time.
Clean start, I've thought. OK, ran ```bbb-install.sh``` again and then get messages, that some files were missing, e.g  ```/etc/bigbluebutton/bigbluebutton-release```. Wondering why, but it was late so I've copied the dpkg-new-files from my last backup. Then the install script ran until end and everything seems fine. But ```bbb-conf --status``` and ```bbb-conf --check``` shows the above output. And BBB didn't worked. 
Then began a long time with searching the failure. In the end I've tried to install all the BBB-packages with ```apt install --reinstall``` but it didn't wrote the missing files. Last step, what brings the solution, was to download the deb-packages and copy the files to ```/etc/bigbluebutton/``` manually. 
So for me the question was why ```apt install --reinstall``` didn't copied the missing files when they weren't there.
